src/ceph.in

```diff
@@ -716,8 +716,8 @@ def ping_monitor(cluster_handle, name, timeout):
         return 1
 
     mon_id = name[len('mon.'):]
+    run_in_thread(cluster_handle.connect, timeout=timeout)
```

Comment:
@trociny i failed to find the analysis or the reasoning of this change. `rados_ping_monitor()` calls `MonClient::ping_monitor()`, which connects to the specified monitor. so why would we call `cluster_handle.connect()` beforehand?

Reply:
It is based on an observation, that for `mon_id == '*'` (working) case we called `cluster_handle.connect()` and for `mon_id != '*'` we did not do it.

I think `cluster_handle.connect()` is needed here to get mon map, so after it the rados "knows" about monitors, while without it it just uses information that it can find in ceph.conf.

In `vstart.sh` cluster we usually have 3 mon hosts specified, and the mon ids happen to be `a`, `b`, `c`, and `ceph ping mon.a` (or b, c) works here.

In a cephadm deployed cluster (I was reproducing the issue on) there was only one mon host in ceph.conf, and interestingly, `ceph ping mon.a` also worked (but not b, c) using the mon_host address from the config, although the mon ID actually was `node1`, not `a`, and `ceph ping mon.node1` did not work.

So, my assumption is that we always need to connect to the cluster to get mon map, so we can find the mon we need. It might violate the purpose of the tool though, if `ceph ping mon.<id>` was supposed to make (check) only the connectivity to this particular monitor. But in this case it have to find all information it needs in ceph.conf and it is not possible. Even if you specify all mon ips in `mon_host` config param, it will not be able to map them to mon ids, so only `ceph ping mon.a|b|c` will work.

Reply:
i see. and i agree that this change practically defeats the purpose of `ceph ping mon.<mon-id>` which is designed to test the reachability of a certain monitor (even if the cluster is not reachable). could you a typical `ceph.conf` generated by `cephadm` deployed cluster ?

Reply:
```
master:~ # cat /etc/ceph/ceph.conf 
# minimal ceph.conf for b103118a-a3d9-11ea-9a2c-5254000d58ac
[global]
        fsid = b103118a-a3d9-11ea-9a2c-5254000d58ac
        mon_host = [v2:10.20.31.201:3300/0,v1:10.20.31.201:6789/0]
master:~ # ceph mon stat
e3: 3 mons at {node1=[v2:10.20.31.201:3300/0,v1:10.20.31.201:6789/0],node2=[v2:10.20.31.202:3300/0,v1:10.20.31.202:6789/0],node3=[v2:10.20.31.203:3300/0,v1:10.20.31.203:6789/0]}, election epoch 572, leader 0 node1, quorum 0,1,2 node1,node2,node3
```
It is deployed with SUSE sesdev [1].

But I think there is nothing special here, because on an older (not `cephadm`) cluster (nautilus), where I have `mon_host` listing all nodes, like this:
```
[global]
fsid = f49c4018-4592-49d4-b2e9-12cb082cbc1e
mon_initial_members = node1, node2, node3
mon_host = 10.20.79.203, 10.20.79.202, 10.20.79.201

e1: 3 mons at {node1=[v2:10.20.79.201:3300/0,v1:10.20.79.201:6789/0],node2=[v2:10.20.79.202:3300/0,v1:10.20.79.202:6789/0],node3=[v2:10.20.79.203:3300/0,v1:10.20.79.203:6789/0]}, election epoch 34, leader 0 node1, quorum 0,1,2 node1,node2,node3
``` 
`ceph ping mon.node1|2|3` does not work, but `ceph ping mon.a|b|c` works.

[1] https://github.com/SUSE/sesdev

Reply:
So if we really do not want cluster connect, we can just document how it works, i.e.:

If one uses `ceph ping mon.<id>`, the `id` is not the mon id but letters `a`, `b`, `c`..., which is the index of the mon adress in `mon_host` list.

Reply:
@trociny i am for updating the document. most background for posterities: it's kind of a dilemma. the `ceph` cli does not have access to a fully populated monmap when serving `ping mon.node1`, while the name of `mon.node1` is actually updated by the monitor of `mon.node1` itself after it boots. BTW, monitor tries to match its `conf->name` with the name listed in `mon_initial_members`. `mon_initial_members` enforces the minimum set of quorum of monitors. so again, it's not related to this use case.

so, what we can use to address the monitors is but "a", "b", ...., instead of their "names".

Reply:
A couple ideas, as this "a", "b" stuff is quite opaque:

* add the "a", "b" information not only to the documentation, but also to the "ping" section of the `ceph` manpage?
* modify the error message that is displayed when the user enters `ceph ping mon.node1` so it briefly explains that `mon.[abc]` will work better?